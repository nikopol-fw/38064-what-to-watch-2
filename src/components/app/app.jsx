import React from 'react';
import PropTypes from 'prop-types';

import MainPage from '../main-page/main-page';
import {FilmPage} from '../film-page/film-page';

const getPageScreen = (props) => {
  const {films, film} = props;

  switch (location.pathname) {
    case `/`:
      return <MainPage films={films}/>;
    case `/films`:
      return <FilmPage
        title={film.title}
        genre={film.genre}
        releaseYear={film.releaseYear}
        posterImg={film.posterImg}
        coverImg={film.coverImg}
      />;
    default:
      return <MainPage films={films}/>;
  }
};


export const App = (props) => {
  return <React.Fragment>{getPageScreen(props)}</React.Fragment>;
};


getPageScreen.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
  })).isRequired,
  film: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseYear: PropTypes.string.isRequired,
    posterImg: PropTypes.string.isRequired,
    coverImg: PropTypes.string.isRequired,
  })
};
