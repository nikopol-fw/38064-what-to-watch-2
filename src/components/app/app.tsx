import * as React  from 'react';
import {Route, Switch} from 'react-router-dom';

import {withLayout} from '../../hocs/with-layout/with-layout';
import MainPage from '../main-page/main-page';
import {FilmPage} from '../film-page/film-page';
import {Login} from '../pages/login/login';


// const getPageScreen = (props) => {
//   const {films, film} = props;
//
//   switch (location.pathname) {
//     case `/`:
//       return ;
//     case `/films`:
//       return <FilmPage
//         title={film.title}
//         genre={film.genre}
//         releaseYear={film.releaseYear}
//         posterImg={film.posterImg}
//         coverImg={film.coverImg}
//       />;
//     default:
//       return <MainPage films={films}/>;
//   }
// };

const MainPageWrapped = withLayout(MainPage);


const App = () => {
  return <Switch>
    <Route path="/" exact component={MainPageWrapped}/>
    <Route path="/login" component={Login}/>
    <Route path="/films/:id" component={FilmPage}/>
  </Switch>;
};


export {App};
