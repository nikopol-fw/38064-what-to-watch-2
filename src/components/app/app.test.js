import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

import {App} from './app';


Enzyme.configure({
  adapter: new Adapter(),
});

it(`App correctly renders after relaunch`, () => {
  const tree = shallow(<App
    user={null}
  />);
  expect(toJSON(tree)).toMatchSnapshot();
});
