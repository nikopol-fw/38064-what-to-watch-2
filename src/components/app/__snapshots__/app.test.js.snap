// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "COUNT_LIKE_THIS": 4,
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(FilmPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/films/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPlayerFunctionality(Player))",
          "type": [Function],
        }
      }
      exact={true}
      path="/films/:id/player"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AddReview)",
          "type": [Function],
        }
      }
      is={null}
      path="/films/:id/review"
      redirectTo="/login"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MyList)",
          "type": [Function],
        }
      }
      data={
        Object {
          "user": null,
        }
      }
      is={null}
      path="/mylist"
      redirectTo="/login"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginPage)",
          "type": [Function],
        }
      }
      is={true}
      path="/login"
      redirectTo="/"
    />
  </Switch>
</Router>
`;
